MirrorProvider/MirrorProvider.Linux/LinuxFileSystemVirtualizer.cs

```diff
@@ -229,19 +229,13 @@ private bool TryGetSymLinkTarget(string relativePath, out string symLinkTarget)
         {
             symLinkTarget = null;
             string fullPathInMirror = this.GetFullPathInMirror(relativePath);
-
-            const ulong BufSize = 4096;
-            byte[] targetBuffer = new byte[BufSize];
-            long bytesRead = ReadLink(fullPathInMirror, targetBuffer, BufSize);
-            if (bytesRead < 0)
```

Comment:
What would you think about leaving the `readlink(2)` code and `DllImport` here in MirrorProvider?  I believe it's used exclusively here and in the [Mac version of MirrorProvider](https://github.com/microsoft/VFSForGit/blob/f6c45002f739e3fa6788a2e83ac1824a34ba91b5/MirrorProvider/MirrorProvider.Mac/MacFileSystemVirtualizer.cs#L254).

Since the syscall is only used in MirrorProvider, not the primary GVFS one, and as it's the same on Darwin/BSD as on Linux, maybe it would be better to pull it out into a `MirrorProvider.POSIX` common class, along with any duplicate code in the `MirrorProvider.Linux/Mac` classes.  I'm thinking that could also be a small separate PR to `features/linuxprototype`.

Reply:
Sounds good! I'll make that PR.